pkgs/by-name/ku/kulala-fmt/package.nix

```diff
@@ -1,29 +1,33 @@
 {
   lib,
-  buildGoModule,
+  buildNpmPackage,
   fetchFromGitHub,
+  nix-update-script,
+  versionCheckHook,
 }:
 
-buildGoModule rec {
+buildNpmPackage rec {
   pname = "kulala-fmt";
-  version = "1.4.0";
+  version = "2.10.0";
 
   src = fetchFromGitHub {
     owner = "mistweaverco";
     repo = "kulala-fmt";
-    rev = "v${version}";
-    hash = "sha256-yq7DMrt+g5wM/tynI7Cf6MBJs/d+fP3IppndKnTJMTw=";
+    tag = "v${version}";
+    hash = "sha256-ICxuNVFHMP8qjOMvWSuestsLWBU3bDVMd1D2TCaugNA=";
   };
 
-  vendorHash = "sha256-GazDEm/qv0nh8vYT+Tf0n4QDGHlcYtbMIj5rlZBvpKo=";
+  postPatch = ''
+    cp ${./package-lock.json} package-lock.json
```

Comment:
Upstream [includes bun.lock](https://github.com/mistweaverco/kulala-fmt/blob/v2.10.0/bun.lock), can't we use it until resolving https://github.com/NixOS/nixpkgs/issues/335534?
https://github.com/NixOS/nixpkgs/blob/affb6b1b6556ee12b94330057a88f8fc344496ff/pkgs/by-name/he/helix-gpt/package.nix#L27

https://github.com/NixOS/nixpkgs/blob/bab385dd73ec557b1c5af0536a78963f1a6175aa/doc/languages-frameworks/javascript.section.md?plain=1#L39-L58